### ISSUE TITLE
Build fails to read environment variable

### ISSUE THREAD

**Issue by postelrich** [conda/conda-build#1815]
Our Jenkins CI started failing this morning.  The job was creating a clean miniconda environment everytime and thus pulling in the latest conda-build version.  Started to get the following error with  conda-build 2.1.6:

```
build/build/myjob_temp/_t_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.5/site-packages/_pytest/junitxml.py", line 376, in pytest_sessionfinish
    os.makedirs(dirname)
  File "/slave1/workspace/myjob/build/myjob_temp/_t_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.5/os.py", line 241, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/build'
```

Looked like it was failing when py.test tries to create a temporary folder for the junit output.  Our `run_test.sh`:

```
#!/usr/bin/env bash
JUNIT=${WORKSPACE}/build
COV=${WORKSPACE}/build

py.test -v --junitxml=$JUNIT/junit.xml --cov myproj --cov-report xml:$COV/coverage.xml tests --spec
```

Which leads me to believe that the `WORKSPACE` envvar wasn't passed in causing py.test to try to make a directory at root. Pegging the version of conda-build to 2.1.5 fixed the problem.

**Comment by msarahan** [comment]
can you show me your recipe?  unless you are explicitly passing the environment variable through with build/script_env ( https://conda.io/docs/building/meta-yaml.html#use-environment-variables ), then it was actually a bug that it was working for you before.

**Comment by postelrich** [comment]
@msarahan no our `meta.yaml` doesn't set the env. So you're saying that conda-build doesn't guarantee to pick up envvars unless explicitly stated in the `meta.yaml`?

**Comment by msarahan** [comment]
Yes.  Is there anything unclear about the docs?  Can we improve them somehow?

> Normally the build script in build.sh or bld.bat does not pass through environment variables from the command line. Only environment variables documented in Environment variables are seen by the build script. To “white-list” environment variables that should be passed through to the build script, use script_env.

**Comment by postelrich** [comment]
No it looks good.  It just always worked for us so we never read that section. I will test it out and see if that resolves the issue with 2.1.6. Thanks!

**Comment by msarahan** [comment]
OK, thanks.  Closing this issue.
